VideoApi/VideoApi.UnitTests/Controllers/MagicLink/AddMagicLinkParticipantTests.cs

```diff
@@ -0,0 +1,100 @@
+using System.Linq;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NUnit.Framework;
+using VideoApi.Common.Security;
+using VideoApi.Contract.Requests;
+using VideoApi.Contract.Responses;
+using VideoApi.DAL.Commands;
+using VideoApi.DAL.Exceptions;
+using VideoApi.DAL.Queries;
+using VideoApi.Extensions;
+using VideoApi.Mappings;
+using Task = System.Threading.Tasks.Task;
+
+namespace VideoApi.UnitTests.Controllers.MagicLink
+{
+    public class AddMagicLinkParticipantTests : MagicLinkControllerTestsBase
+    {
+        [Test]
+        public async Task Should_call_query_handler_to_getConference()
+        {
+            //Arrange/Act
+            await Controller.AddMagicLinkParticipant(HearingId, AddMagicLinkParticipantRequest);
+
+            //Assert
+            QueryHandler.Verify(x => x.Handle<GetConferenceByHearingRefIdQuery, VideoApi.Domain.Conference>(
+                It.IsAny<GetConferenceByHearingRefIdQuery>()), Times.Once);
+        }
+
+        [Test]
+        public async Task Should_return_not_found_result_ifConference_is_not_found()
+        {
+            //Arrange
+            QueryHandler.Setup(x => x.Handle<GetConferenceByHearingRefIdQuery, VideoApi.Domain.Conference>(
+                It.IsAny<GetConferenceByHearingRefIdQuery>())).ThrowsAsync(new ConferenceNotFoundException(HearingId));
+
+            //Act
+            var result = await Controller.AddMagicLinkParticipant(HearingId, AddMagicLinkParticipantRequest) as NotFoundObjectResult;
+
+            //Assert
+            Assert.IsInstanceOf<NotFoundObjectResult>(result);
+            Assert.False((bool)result.Value);
+        }
+
+        [Test]
+        public async Task Should_call_command_handler_to_add_participant_toConference()
+        {
+            //Arrange/Act
+            await Controller.AddMagicLinkParticipant(HearingId, AddMagicLinkParticipantRequest);
+
+            //Assert
+            CommandHandler.Verify(x => x.Handle(It.Is<AddParticipantsToConferenceCommand>(x =>
```

Comment:
I dont think we need to test implementation, I would remove this test as we dont want to check if a certain function is being called. However we can do this instead, we can use this as a setup to return a specific value that will eventually return a success response and assert that.